From 02cb19835cb52bd84b0b5eaca1b4d6338417d261 Mon Sep 17 00:00:00 2001
From: Antoine Jacoutot <ajacoutot@gnome.org>
Date: Tue, 15 Oct 2013 07:36:30 +0000
Subject: portability: canonicalize_file_name -> realpath

canonicalize_file_name() is a GNU extension so let's use the portable
realpath() instead.

https://bugzilla.gnome.org/show_bug.cgi?id=710163
---
diff --git a/rsvg-base.c b/rsvg-base.c
index cb9f64a..fb829e7 100644
--- a/rsvg-base.c
+++ b/rsvg-base.c
@@ -50,6 +50,8 @@
 #include <math.h>
 #include <string.h>
 #include <stdarg.h>
+#include <limits.h>
+#include <stdlib.h>
 
 #include "rsvg-path.h"
 #include "rsvg-paint-server.h"
@@ -2190,8 +2192,7 @@ _rsvg_handle_allow_load (RsvgHandle *handle,
     dir = g_file_get_path (base);
     g_object_unref (base);
 
-    /* FIXME portability */
-    cdir = canonicalize_file_name (dir);
+    cdir = realpath (dir, NULL);
     g_free (dir);
     if (cdir == NULL)
         goto deny;
@@ -2200,8 +2201,7 @@ _rsvg_handle_allow_load (RsvgHandle *handle,
     if (path == NULL)
         goto deny;
 
-    /* FIXME portability */
-    cpath = canonicalize_file_name (path);
+    cpath = realpath (path, NULL);
     g_free (path);
 
     if (cpath == NULL)
--
cgit v0.9.2
